spline-shadow/client-ui/src/app/components/lineage/lineage-details/schema-table/schema-table.component.ts

```diff
@@ -13,78 +13,94 @@
+  encapsulation: ViewEncapsulation.None
 })
-export class SchemaTableComponent implements OnInit {
-
-  @ViewChild('detailsLine', { read: ViewContainerRef })
-  private _detailsLine: ViewContainerRef
-
-  dtOptions: DataTables.Settings = {}
+export class SchemaTableComponent implements AfterViewInit {
 
-  selectedItemName: any
+  @ViewChild('table')
+  table: any
 
-  expandedRows: Set<any> = new Set()
+  expanded: any = {}
 
   @Input()
   schema: any
 
+  @Input()
+  schemaId: string
+
+  tablePageSize: number = 5
+
   constructor(
-    private componentFactoryResolver: ComponentFactoryResolver,
-    private propertyService: PropertyService
+    private propertyService: PropertyService,
+    private routerService: RouterService
   ) { }
 
-  ngOnInit() {
-    this.dtOptions = {
-      pagingType: 'simple_numbers',
-      pageLength: 10,
-      lengthChange: false,
-      searching: false,
-      info: false,
-      order: [[1, 'asc']],
-      preDrawCallback: function (settings: any) {
-        let api = new $.fn.dataTable.Api(settings)
-        let pagination = $(this).closest('.dataTables_wrapper').find('.dataTables_paginate')
-        pagination.toggle(api.page.info().pages > 1)
+  ngAfterViewInit(): void {
+    const self = this
+    this.routerService.getParams().subscribe(config => {
+      const that = this
+      const schemaIdParam = config.get("schemaId")
+      const tablesWithSelection = schemaIdParam ? schemaIdParam.split(".") : []
+      if (that.table.rows && that.schemaId.includes(tablesWithSelection[0])) {
+        for (let i = 0; i < tablesWithSelection.length + 1; i++) {
+          //The property ID can be nested to several data structures, the last one is the selected property itself
+          let propertyIdParam = (i < tablesWithSelection.length) ? tablesWithSelection[i + 1] : config.get("property")
+          let selectedRow = that.getSelectedRowFromName(propertyIdParam)
+          if (selectedRow) {
+            that.propertyService.changeCurrentProperty(selectedRow[1])
+            that.table.selected.push(selectedRow[1])
+            let page = Math.floor(selectedRow[0] / that.tablePageSize)
+            that.table.offset = page
+            // TODO : Remove the setTimeout as soon as this issue is fixed :https://github.com/swimlane/ngx-datatable/issues/1204
+            setTimeout(function () {
+              if (selectedRow[1].dataType._typeHint != 'za.co.absa.spline.core.model.dt.Simple') {
+                that.table.rowDetail.toggleExpandRow(selectedRow[1])
+              }
+            })
+          }
+        }
       }
-    }
+    })
   }
 
-  rowClick(item: any) {
-    this.propertyService.changeCurrentProperty(item)
-    this.selectedItemName = item.name
+  /**
+   * Gets selected row from name
+   * @param name the name of the property
+   * @returns a tuple containing the row itself and it's index in case the table is pageable
+   */
+  getSelectedRowFromName(name) {
+    let selectedRow = null
```

Comment:
Looks like the method is just `const i = _.findLastIndex(rows, {name: name}); return [i, rows[i]]`.
If in fact there is no duplicated row names then using `_.findIndex` will be faster as it terminates earlier.